src/Processors/QueryPlan/Optimizations/optimizeTree.cpp

```diff
@@ -317,7 +317,7 @@ void optimizeTreeSecondPass(
                 }
 
                 if (optimization_settings.aggregation_in_order)
-                    optimizeAggregationInOrder(*frame.node, nodes);
+                    optimizeAggregationInOrder(*frame.node, nodes, optimization_settings);
```

Comment:
Is `optimizeAggregationInOrder` supported with JOINs as well?
(good thing to to as well)